exec/kubernetes_exec_manager.go

```diff
@@ -341,11 +341,13 @@ func (manager *KubernetesExecManager) CreateKubeConfig(initConfigParams *model.I
 	for _, containerInfo := range containersInfo {
 		if containerInfo.ContainerName == initConfigParams.ContainerName || initConfigParams.ContainerName == "" {
 			namespace := GetNamespace()
+			infoExecCreator := exec_info.NewKubernetesInfoExecCreator(namespace, k8sAPI.GetClient().Core(), k8sAPI.GetConfig())
+
+			userDefinedNamespace := namespace
 			if initConfigParams.Namespace != "" {
-				namespace = initConfigParams.Namespace
+				userDefinedNamespace = initConfigParams.Namespace
 			}
-			infoExecCreator := exec_info.NewKubernetesInfoExecCreator(namespace, k8sAPI.GetClient().Core(), k8sAPI.GetConfig())
-			err = kubeconfig.CreateKubeConfig(infoExecCreator, namespace, &initConfigParams.KubeConfigParams, containerInfo)
+			err = kubeconfig.CreateKubeConfig(infoExecCreator, userDefinedNamespace, &initConfigParams.KubeConfigParams, containerInfo)
```

Comment:
For me, it's not clear why CreateKubeConfig accepts namespace parameter when another argument - kubeconfigParam already has that field. Also, current namespace is not going to be changed. 
So, I propose the following change:

```go
	currentNamespace := GetNamespace()
	infoExecCreator := exec_info.NewKubernetesInfoExecCreator(currentNamespace, k8sAPI.GetClient().Core(), k8sAPI.GetConfig())

	if initConfigParams.Namespace == "" {
		initConfigParams.Namespace = currentNamespace
	}

	for _, containerInfo := range containersInfo {
		if containerInfo.ContainerName == initConfigParams.ContainerName || initConfigParams.ContainerName == "" {
			err = kubeconfig.CreateKubeConfig(infoExecCreator, &initConfigParams.KubeConfigParams, containerInfo)
			if err != nil {
				return err
			}
			return nil
		}
	}
```